tests/common/runner.rs

```diff
@@ -66,7 +66,7 @@ pub fn test_execute_alloy<S: Storage + Send + Sync, C: PevmChain + Send + Sync +
         // Before EIP-658 (https://eips.ethereum.org/EIPS/eip-658), the
         // receipt root is calculated with the post transaction state root,
         // which we don't have in these tests.
-        if block.header.number >= 4370000 {
+        if spec_id >= SpecId::BYZANTIUM {
```

Comment:
This temporary fix will only last to the OP PR as we parameterize this function by `Chain` but most chains don't have the concept of `BYZANTIUM`. The better fix (for now) is to make `calculate_receipt_root` return an `Option` and only make the assertion if it returns a `Some` here. We then move this EIP-658 logic to Ethereum's implementation of `calculate_receipt_root` and return a `None` (currently unsupported) pre-BYZANTIUM. New chain users can also return a `None` until/unless they need a receipt test/integration.

Reply:
I changed the function signature to:

```rust
    /// Calculate receipt root
    fn calculate_receipt_root(
        &self,
        spec_id: SpecId,
        txs: &BlockTransactions<Self::Transaction>,
        tx_results: &[PevmTxExecutionResult],
    ) -> Result<Option<B256>, Self::CalculateReceiptRootError>;
```

Can you help to review again? thanks!